tessera-jaxrs/thirdparty-jaxrs/src/main/java/com/quorum/tessera/thirdparty/PartyInfoResource.java

```diff
@@ -0,0 +1,53 @@
+package com.quorum.tessera.thirdparty;
+
+import com.quorum.tessera.partyinfo.PartyInfoService;
+import com.quorum.tessera.partyinfo.model.PartyInfo;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import static java.util.Objects.requireNonNull;
+
+@Path("/partyinfo")
+public class PartyInfoResource {
+
+    private final PartyInfoService partyInfoService;
+
+    public PartyInfoResource(final PartyInfoService partyInfoService) {
+        this.partyInfoService = requireNonNull(partyInfoService, "partyInfoService must not be null");
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Fetch network/peer information", produces = "public list of peers/publickey mappings")
+    @ApiResponses({@ApiResponse(code = 200, message = "Peer/Network information", response = PartyInfo.class)})
+    public Response getPartyInfo() {
+
+        final PartyInfo current = this.partyInfoService.getPartyInfo();
+
+        final JsonArrayBuilder recipientBuilder = Json.createArrayBuilder();
+        current.getRecipients().stream()
+                .map(
+                        recipient ->
+                                Json.createObjectBuilder()
+                                        .add("key", recipient.getKey().encodeToBase64())
```

Comment:
Would it be better to return a simple array of public keys as opposed to an array of single field objects?  It would help to make the API easier to document

Reply:
The main reason to do it like this is to future-proof it. We can add more details to the object without breaking users that already rely on this API.

Reply:
Sure - alternatively the endpoint could be `/partyinfo/keys` or something similarly specific, thereby leaving `/partyinfo` free for future use.  I'm not fussed either way but figured the question should be asked.